src/Simple.OData.Client.Core/Adapter/BatchWriterBase.cs

```diff
@@ -86,7 +98,7 @@ public void MapContentId(IDictionary<string, object> entryData, string contentId
 
         public IDictionary<object, IDictionary<string, object>> BatchEntries { get; private set; }
 
-        public async Task<object> CreateOperationMessageAsync(Uri uri, string method, string collection, IDictionary<string, object> entryData, bool resultRequired)
+        public async Task<object> CreateOperationMessageAsync(Uri uri, string method, string collection, IDictionary<string, object> entryData, IDictionary<string, string> operationHeaders, bool resultRequired)
```

Comment:
This is a breaking change to a public class. A method should be left with the old signature that forwards to this new public signature (with the appropriate value for `operationHeaders`).

Reply:
I spent actually quite some time with the suggested approach but didnt really find a good solution to this. The primary reason being that this method is invoked from other methods up in the call chain. And since the operationHeaders is sent all the way from the command that seemed to be the most viable options. 
If you take a closer look i have therefore updated all the references of this method with a null value in the places where it is being called. Unit tests updated accordingly.
This of course given the fact that i was just introduced to this lib. Hence I rely on some more input from someone who knows the lib inside out :)